src/setupEnzyme.tsx

```diff
@@ -22,15 +22,15 @@ import { RemainingPropsAndTestOverrides } from "./types";
  */
 export function setupEnzyme<
   ComponentType extends React.ComponentType,
-  BaseProps extends Partial<React.ComponentProps<ComponentType>>
->(Component: ComponentType, baseProps?: BaseProps) {
+  BaseProps extends Partial<FullProps<ComponentType>> = {}
```

Comment:
This line throws a linter error, however the default type of an empty object _is_ what we want. It signifies to the rest of the type engine that if no second param is provided in the initial `setup*` call, that the interface to use is an empty object. From here, our union types work properly to cobble together the remaining required props.

Reply:
Yeah you can just disable that in `.eslintrc.js` if you want. Seems reasonable how you have it.

Reply:
Cool. I've opted to go with a single-line linter disable (since I think it's generally a good rule, and actually caught me in another area where I was able to use the `Record` type instead. Glad to get this feedback though =D

Reply:
nice!